libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/RouteLineComponent.kt

```diff
@@ -10,61 +10,37 @@ import com.mapbox.maps.plugin.gestures.gestures
 import com.mapbox.maps.plugin.locationcomponent.OnIndicatorPositionChangedListener
 import com.mapbox.maps.plugin.locationcomponent.location
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
-import com.mapbox.navigation.base.route.NavigationRoute
 import com.mapbox.navigation.core.MapboxNavigation
 import com.mapbox.navigation.core.internal.extensions.flowRouteProgress
-import com.mapbox.navigation.core.internal.extensions.flowRoutesUpdated
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
 import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import com.mapbox.navigation.ui.maps.internal.ActiveRouteLineComponentContract
+import com.mapbox.navigation.ui.maps.internal.RouteLineComponentContract
 import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.findClosestRoute
 import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.setNavigationRouteLines
 import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
 import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
 import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
 import com.mapbox.navigation.ui.maps.route.line.model.NavigationRouteLine
-import com.mapbox.navigation.ui.utils.internal.Provider
 import com.mapbox.navigation.utils.internal.ifNonNull
-import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 
-@ExperimentalPreviewMapboxNavigationAPI
-interface RouteLineComponentContract {
-    fun setRoutes(mapboxNavigation: MapboxNavigation, routes: List<NavigationRoute>)
-}
-
-@ExperimentalPreviewMapboxNavigationAPI
-internal class MapboxRouteLineComponentContract : RouteLineComponentContract {
```

Comment:
Why did we remove this implementation and it's use from everywhere?

Reply:
This wasn't removed, it was moved to a separate file. This contract of, observing and updating `MapboxNavigation` is also needed by androidauto components. 

So I've called it ActiveRouteLineComponentContract
to have a distinction from PreviewRouteLineComponentContract